src/librustc_mir/interpret/memory.rs

```diff
@@ -73,26 +73,26 @@ impl MemoryPointer {
         MemoryPointer { alloc_id, offset }
     }
 
-    pub fn wrapping_signed_offset<'tcx>(self, i: i64, layout: &TargetDataLayout) -> Self {
-        MemoryPointer::new(self.alloc_id, value::wrapping_signed_offset(self.offset, i, layout))
+    pub(crate) fn wrapping_signed_offset<'a, L: HasDataLayout<'a>>(self, i: i64, l: L) -> Self {
```

Comment:
FWIW there are a bunch of things in miri that can now be replaced with [some trans-like setup](https://github.com/rust-lang/rust/blob/ef19bd1a339212f77160f12af6d5d85d8bb36ea4/src/librustc_trans/context.rs#L729-L777) plus using the `TyLayout` methods - and they'll be guaranteed to work in any situation the trans code already does.